### PR TITLE
Programlist

### PR DESCRIPTION
**Pull request by snschau**
To create a page to display all the programs 

FRONTED (Gatsby): https://gusprogramlist.gatsbyjs.io/programs/

- programs.js - creates a page that has the programs listed in alphabetical order. Using cards linking back to the program page. 
- programsList.css - css for the programs list page

BACKEND (Drupal): live
- no changes

TESTING: 
The page https://gusprogramlist.gatsbyjs.io/programs/ should display all the program pages that are published - in alphabetical order.

### CODE REVIEW

**Comment by tqureshi-uog** [comment]
@snschau I'm seeing a bug that breaks the `Back` button. Steps to reproduce:

1. Go to https://gusprogramlist.gatsbyjs.io/programs/
2. Click any program
3. Click the `Back` button

**Expected behavior:** You'll be returned to the programs list page.
**Observed behavior:** The URL in the address bar changes to https://gusprogramlist.gatsbyjs.io/programs/ but the content for whatever program you clicked before remains onscreen. This happens in Chrome, Edge, and Firefox.

This is a known issue: https://stackoverflow.com/questions/54856449/browser-back-button-not-working-for-gatsby-js-project

Not sure why it's happening for the Programs list page but not others. One possible fix to try is using regular `<a href` tags instead of Gatsby's `Link` plugin.

**Comment by snschau** [comment]
> 

Fixed - backbutton works.

File: src/pages/programs.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    <Layout>
+      {data.allNodeProgram.edges.sort(function(a,b){
+        var titleA =a.node.title, titleB = b.node.title;
+        if (titleA < titleB)
+          return -1;
+        if (titleA > titleB)
+          return 1;
+        return 0;
+      })}
+        <Seo title="Program List" keywords={[`gatsby`, `application`, `react`]} />
```
Comment (tqureshi-uog) [inline src/pages/programs.js:17]:
```suggestion
        <Seo title="Program List" />
```

The `keywords` field is no longer recognized by reputable search engines. We could replace it with `description="whatever"` but someone else (either from Admissions or C&PA) would need to decide how this Program Listing page should be described.

Comment (snschau) [reply src/pages/programs.js:17]: will bring it up at the huddle tomorrow

File: src/styles/programsList.css [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    border-left: 1rem solid var(--uog-yellow);
+  }
+  .programsList-card.card:nth-of-type(12n+3),
+  .programsList-card.card:nth-of-type(12n+6),
+  .programsList-card.card:nth-of-type(12n+9),
+  .programsList-card.card:nth-of-type(12n+12) {
+    border-left: 1rem solid var(--uog-blue);
+}
+  
+  .programsList-card:hover {
```
Comment (tqureshi-uog) [inline src/styles/programsList.css:44]:
```suggestion
  .programsList-card:hover,
  .programsList-card:focus {
```

So keyboard users also see the grey background when they tab to a program card.

File: src/styles/programsList.css [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    border-left: 1rem solid var(--uog-blue);
+}
+  
+  .programsList-card.card:hover,
+  .programsList-card.card:focus {
+    background-color: #d3d3d3 !important;
+  }
+
+  .page-container a.stretched-link:hover,
+  .page-container a.streched-link:focus {
```
Comment (tqureshi-uog) [inline src/styles/programsList.css:50]:
```suggestion
  .page-container a.stretched-link:focus {
```
 
Typo fix - the t was missing in "stretched"

**Review by tqureshi-uog — APPROVED** [review]

**Pull request merged**